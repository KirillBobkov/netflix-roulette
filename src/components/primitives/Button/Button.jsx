import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

export const Button = (props) =>  (
  <button
    onClick={props.onClick}
    className={'button ' + `${props.className || ''}`}
    id={`${props.id || ''}`}
  >
    {props.text}
  </button>
);

Button.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string,
  id: PropTypes.string,
  onClick: PropTypes.func
};
