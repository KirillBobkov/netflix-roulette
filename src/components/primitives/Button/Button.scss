.button {
    color: #ffffff;
    border: none;
    outline: 1px solid transparent;
    text-transform: uppercase;
    padding: 6px 17px;
    background-color: #272727b7; 
    transition: all 0.5s ease-out; 
    cursor: pointer;

        &--reset {
            position: absolute;
            top: 9px;
            right: 205px;
            padding: 6px;
        }

        &--search {
            border-radius: 3px;
            background: #F65261;
            padding: 15px 15px 13px;
            width: 188px;
            margin-left: 10px;
            font-size: 16px;
            line-height: 16px;

            &:active {
                background-color: #272727b7; 
            }
        }

        &--choosen {
            background: #F65261;
            padding: 6px 17px;
        }

        &--left-border {
            border-top-left-radius: 3px;
            border-bottom-left-radius: 3px;
        }

        &--right-border {
            border-top-right-radius: 3px;
            border-bottom-right-radius: 3px;
        }

        &--search-icon {
            position: absolute;
            top: 2px;
            right: 5px;
            display: block;
            width: 40px;
            height: 40px;
            border-radius: 20px;
            background-image: url('../../../assets/images/searchIcon.png');
            background-size: cover;
            background-color: transparent;  
        }

}
