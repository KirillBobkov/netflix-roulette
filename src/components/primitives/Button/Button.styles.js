import styled from "styled-components";

export const ButtonInnerWrapper = styled.button`
  color: #ffffff;
  border: none;
  outline: 1px solid transparent;
  text-transform: uppercase;
  padding: 6px 17px;
  background-color: #272727b7; 
  transition: all 0.5s ease-out; 
  cursor: pointer;

  ${props => {
    if (props.reset) {
      return `
          position: absolute;
          top: 9px;
          right: 205px;
          padding: 6px;
      `;
    }
  }}

  ${props => {
    if (props.search) {
      return `
        border-radius: 3px;
        background: #F65261;
        padding: 15px 15px 13px;
        width: 188px;
        margin-left: 10px;
        font-size: 16px;
        line-height: 16px;

        &:active {
            background-color: #272727b7; 
        }
      `;
    }
  }}

  ${props => {
    if (props.choosen) {
      return `
        background: #F65261;
        padding: 6px 17px;
      `;
    }
  }}

  ${props => {
    if (props.leftBorder) {
      return `
        border-top-left-radius: 3px;
        border-bottom-left-radius: 3px;
      `;
    }
  }}

  ${props => {
    if (props.rightBorder) {
      return `
        border-top-right-radius: 3px;
        border-bottom-right-radius: 3px;
      `;
    }
  }}

  ${props => {
    if (props.searchIcon) {
      return `
        position: absolute;
        top: 10px;
        right: 17px;
        display: block;
        width: 20px;
        height: 20px;
        padding: 0;
        border-radius: 20px;
        border: 4px solid #F65261;
        background-size: cover;
        background-color: transparent;  

        &:after {
            position: absolute;
            top: 14px;
            right: -9px;
            content: '';
            display: inline-block;
            width: 10px;
            height: 4px;
            background: #F65261;
            transform: rotate(45deg);
        }
      `;
    }
  }}
`;