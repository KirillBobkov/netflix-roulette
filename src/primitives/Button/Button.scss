.button {
    color: #ffffff;
    border: none;
    outline: 1px solid transparent;
    text-transform: uppercase;
    padding: 6px 17px;
    background-color: #272727b7;  

        &--search {
            border-radius: 3px;
            background: #F65261;
            padding: 15px;
            width: 188px;
            margin-left: 10px;
            margin-bottom: 10px;
        }

        &--choosen {
            background: #F65261;
            padding: 6px 17px;
            font-size: 12px;
        }

        &--left-border {
            border-top-left-radius: 3px;
            border-bottom-left-radius: 3px;
        }

        &--right-border {
            border-top-right-radius: 3px;
            border-bottom-right-radius: 3px;
        }

}
