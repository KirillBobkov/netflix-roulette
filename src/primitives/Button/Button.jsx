import React from 'react';
import './Button.scss';

export function Button(props) {
  return (
    <button
      onClick={props.onClick}
      className={'button ' + `${props.className || ''}`}
      id={`${props.id || ''}`}
    >
      {props.text}
    </button>
  );
}
