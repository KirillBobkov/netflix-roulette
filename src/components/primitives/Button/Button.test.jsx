import React from 'react';
import { Button } from './Button';
import { shallow, mount } from 'enzyme';
import { config } from 'chai';

describe('Button component', () => {
    let config, 
    componentShallowed,
    componentMounted,
    handleClick;

    beforeAll(() => {
        handleClick = jest.fn();
        config = {
            class: 'button button--search',
            text: 'Search'
        };
        componentShallowed = shallow(<Button className={config.class} text={config.text} onClick={handleClick} /> );
        componentMounted = mount(<Button className={config.class} text={config.text} onClick={handleClick} /> );
    }); 
   
    it('should be render correctly', () => {
        expect(componentShallowed).toMatchSnapshot();
    });

    it('should check that inner text of Button', () => {
        expect(componentMounted.text()).toEqual('Search');
    });

    it('should be call onClick', () => {
        componentMounted.simulate('click');
        expect(handleClick).toHaveBeenCalled();
    });
});