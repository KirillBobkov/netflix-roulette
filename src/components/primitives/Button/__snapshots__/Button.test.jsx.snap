// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should be render correctly 1`] = `ShallowWrapper {}`;
